pkg/healthchecks/healthchecks.go

```diff
@@ -62,6 +62,10 @@ const (
 	// backends, the port or named port specified in the Backend Service is
 	// used for health checking.
 	UseServingPortSpecification = "USE_SERVING_PORT"
+
+	newHealthCheckErrorMessageTemplate = "the %v health check configuration on the exiting health check %v is nil. " +
+		"This is usually caused by application protocol changed. " +
+		"Please revert the change on application protocol to avoid this error message."
```

Comment:
"...revert the change on the application..."

Reply:
```suggestion
		"Please revert the change on the application's protocol to avoid this error message."
```

I don't have a lot of insight into how the ingress-gce works internally, but I think a TODO on handling this type of change would be helpful. I think progressing from HTTP-1.1 to HTTP-2 (soon HTTP-3) or introducing changes on the protocol to adopt a service mesh will be a common occurrence.

Reply:
yeah. Agreed. In the long run we want to support this. But currently, due to GCE api changes, it was broken. So we need to put this short term fix in place so that it can be cherry picked.

Reply:
Can we add the TODO just so we remember to do it?